packages/google_maps_flutter/google_maps_flutter/lib/src/controller.dart

```diff
@@ -38,53 +46,108 @@ class GoogleMapController {
-        .onTap(mapId: mapId)
-        .listen((MapTapEvent e) => _googleMapState.onTap(e.position));
-    GoogleMapsFlutterPlatform.instance.onLongPress(mapId: mapId).listen(
-        (MapLongPressEvent e) => _googleMapState.onLongPress(e.position));
-    GoogleMapsFlutterPlatform.instance
-        .onClusterTap(mapId: mapId)
-        .listen((ClusterTapEvent e) => _googleMapState.onClusterTap(e.value));
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance
+          .onMarkerTap(mapId: mapId)
+          .listen((MarkerTapEvent e) => _googleMapState.onMarkerTap(e.value)),
+    );
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance.onMarkerDragStart(mapId: mapId).listen(
+            (MarkerDragStartEvent e) =>
+                _googleMapState.onMarkerDragStart(e.value, e.position),
+          ),
+    );
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance.onMarkerDrag(mapId: mapId).listen(
+            (MarkerDragEvent e) =>
+                _googleMapState.onMarkerDrag(e.value, e.position),
+          ),
+    );
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance.onMarkerDragEnd(mapId: mapId).listen(
+            (MarkerDragEndEvent e) =>
+                _googleMapState.onMarkerDragEnd(e.value, e.position),
+          ),
+    );
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance.onInfoWindowTap(mapId: mapId).listen(
+            (InfoWindowTapEvent e) => _googleMapState.onInfoWindowTap(e.value),
+          ),
+    );
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance.onPolylineTap(mapId: mapId).listen(
+            (PolylineTapEvent e) => _googleMapState.onPolylineTap(e.value),
+          ),
+    );
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance.onPolygonTap(mapId: mapId).listen(
+            (PolygonTapEvent e) => _googleMapState.onPolygonTap(e.value),
+          ),
+    );
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance
+          .onCircleTap(mapId: mapId)
+          .listen((CircleTapEvent e) => _googleMapState.onCircleTap(e.value)),
+    );
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance
+          .onTap(mapId: mapId)
+          .listen((MapTapEvent e) => _googleMapState.onTap(e.position)),
+    );
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance.onLongPress(mapId: mapId).listen(
+            (MapLongPressEvent e) => _googleMapState.onLongPress(e.position),
+          ),
+    );
+    _addSubscription(
+      GoogleMapsFlutterPlatform.instance.onClusterTap(mapId: mapId).listen(
+            (ClusterTapEvent e) => _googleMapState.onClusterTap(e.value),
+          ),
+    );
+  }
+
+  /// Adds a stream subscription to the list of active subscriptions.
+  ///
+  /// This method is used to track and manage all stream subscriptions
+  /// created for map events. The subscriptions are stored in
+  /// [_streamSubscriptionList] and will be properly disposed when the
+  /// controller is disposed.
```

Comment:
This second sentence should be removed, as it is asserting things outside of the control of this method.

In general, it is an anti-pattern for a comment in one part of the code to make assertions about *other code*, because the chances of someone changing code updating all assertions about that code anywhere else in the codebase are extremely low, making them highly prone to becoming outdated and misleading.